posts/2021/09/array-libraries-interoperability.md

```diff
@@ -0,0 +1,578 @@
+<!--
+.. title: Array Libraries Interoperability
+.. slug: array-libraries-interoperability
+.. date: 2021-09-20 17:04:54 UTC-05:00
+.. author: Anirudh Dagar
+.. tags: SciPy, PyTorch, NumPy, CuPy, uarray, Array API
+.. category: 
+.. link: 
+.. description: 
+.. type: text
+.. previewimage: /images/2021/09/ninja_arrays.png
+-->
+
+In this blog post I talk about the work that I was able to accomplish during
+my internship at Quansight Labs and the efforts being made towards making
+array libraries more interoperable.
+
+Going ahead, I'll assume basic understanding of array and tensor libraries
+with their usage in the Python Scientific and Data Science software stack.
+
+<p align="center">
+	<img alt="Ninja Turtles Array Libraries" src="/images/2021/09/ninja_arrays.png">
+    <i>Old Ninja Turtle NumPy and young Tensor Turtles</i>
+</p>
+
+<!-- TEASER_END -->
+
+## Array Provider & Consumer Libraries
+
+As someone who's been involved in the Data Science domain for more than four
+years now, I can't think of any of my projects without using an array/tensor
+library. Over time, NumPy (more than 15 years old today) has become a blind
+import before starting any Python project involving manipulation of arrays.
+
+Quite recently with the Deep Learning buzz, more and more libraries have been
+developed with some extra sweetness on top of the NumPy like `nd-array`
+computation functionality. Namely this extra sweetness is the multi-device
+(`CPU`, `GPU`, `XLA` etc.) support and autograd engine for the array/tensor
```

Comment:
Slightly rephrased this. I believe N-dimensional arrays is more common and slightly compact. Also goes well with the "ndarray" instead of "multi-dimensional" array. Both are correct of course.